### PR TITLE
fix: discard changes of timezone, locale and dark mode reset initial …

### PR DESCRIPTION
**Pull request by CataldoMazzilli**
…values

refs: SHELL-161

### CODE REVIEW

**Review by nubsthead — DISMISSED** [review]

File: src/settings/language-and-timezone-settings.tsx [hunk 1]
```diff
@@ -4,74 +4,95 @@
...
-	const defaultLocale = useMemo(
-		() =>
-			(settings.prefs.zimbraPrefLocale &&
-				find(locales, { id: `${settings.prefs.zimbraPrefLocale}` })) ||
-			find(locales, { id: 'en' }),
-		[locales, settings.prefs.zimbraPrefLocale]
+	const [prefLocale, setPrefLocale] = useState<string>(settings.prefs.zimbraPrefLocale ?? '');
+
+	const prefLocaleSelectedValue = useMemo<SelectItem>(
+		() => find(locales, (item) => item.value === prefLocale) as SelectItem,
```
Comment (beawar) [inline src/settings/language-and-timezone-settings.tsx:56]: Can we add a comment here with the ref of the issue with which we will remove the cast?

File: src/settings/language-and-timezone-settings.tsx [hunk 2]
```diff
@@ -4,74 +4,95 @@
...
-			(settings.prefs.zimbraPrefTimeZoneId &&
-				find(timezones, { value: `${settings.prefs.zimbraPrefTimeZoneId}` })) ||
-			find(timezones, { value: 'UTC' }),
-		[timezones, settings.prefs.zimbraPrefTimeZoneId]
+	const [prefTimeZoneId, setPrefTimeZoneId] = useState<string>(
+		settings.prefs.zimbraPrefTimeZoneId ?? ''
+	);
+
+	const prefTimeZoneIdSelectedValue = useMemo<SelectItem>(
+		() => find(timezones, (item) => item.value === prefTimeZoneId) as SelectItem,
```
Comment (beawar) [inline src/settings/language-and-timezone-settings.tsx:76]: Can we add a comment with the issue ref?

File: src/settings/language-and-timezone-settings.tsx [hunk 3]
```diff
@@ -80,31 +101,28 @@ const LanguageAndTimeZone: FC<{
...
+					background={'gray5'}
+					label={t('label.language', 'Language')}
+					onChange={onLocaleChange}
+					selection={prefLocaleSelectedValue}
+					showCheckbox={false}
+					dropdownMaxHeight="12.5rem"
+					selectedBackgroundColor="highlight"
+				/>
+
+				<Padding top="small" width="100%">
```
Comment (beawar) [inline src/settings/language-and-timezone-settings.tsx]: we could remove this padding and use the container gap instead

**Review by beawar — COMMENTED** [review]
Discard and save buttons remain enabled also when, after changing some selection, I reset everything to the previous situation.
I've checked on account settings and I think we never defined how the UI should behave in this situation, because some settings are disabling the buttons once the user set everything as it was, some others are leaving the buttons enabled. I feel we need to establish a single common behavior

**Comment by CataldoMazzilli** [comment]
> Discard and save buttons remain enabled also when, after changing some selection, I reset everything to the previous situation. I've checked on account settings and I think we never defined how the UI should behave in this situation, because some settings are disabling the buttons once the user set everything as it was, some others are leaving the buttons enabled. I feel we need to establish a single common behavior

ok created specific issue

**Review by beawar — APPROVED** [review]

**Review by nubsthead — APPROVED** [review]

**Review by nubsthead — APPROVED** [review]

**Pull request merged**